app/command_code.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"strings"
+
+	"github.com/mattermost/platform/model"
+	goi18n "github.com/nicksnyder/go-i18n/i18n"
+)
+
+type CodeProvider struct {
+}
+
+const (
+	CMD_CODE = "code"
+)
+
+func init() {
+	RegisterCommandProvider(&CodeProvider{})
+}
+
+func (me *CodeProvider) GetTrigger() string {
+	return CMD_CODE
+}
+
+func (me *CodeProvider) GetCommand(T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_CODE,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_code.desc"),
+		AutoCompleteHint: T("api.command_code.hint"),
+		DisplayName:      T("api.command_code.name"),
+	}
+}
+
+func (me *CodeProvider) DoCommand(args *model.CommandArgs, message string) *model.CommandResponse {
+	rmsg := ""
+	if len(message) > 0 {
```

Comment:
I don't think we should have an error if the message is empty. We don't typically allow users to make empty posts